packages/http-server/src/types.ts

```diff
@@ -16,6 +15,4 @@ export interface IPrismHttpServer {
   listen: (port: number, address?: string, backlog?: number) => Promise<string>;
 }
 
-export type ThenArg<T> = T extends Promise<infer U> ? U :
-  T extends ((...args: any[]) => Promise<infer U>) ? U :
-    T;
+export type ThenArg<T> = T extends Promise<infer U> ? U : T extends (...args: any[]) => Promise<infer U> ? U : never;
```

Comment:
When I saw it I was afraid that you did some changes to that line. Fortunately, it is linter who did it ;)